tests/test_dev_server.py

```diff
@@ -49,10 +48,87 @@ def test_my_lambda_dev_handler():
+
+@pytest.mark.parametrize(
+    "method", ["do_GET", "do_PATCH", "do_POST", "do_PUT", "do_DELETE", "do_OPTIONS"]
+)
+def test_handle_request(method) -> None:
+    with patch("socket.socket") as msocket:
+        msocket.makefile = lambda a, b: io.BytesIO(b"GET / HTTP/1.1\r\n")
+        msocket.rfile.close = lambda: 0
+        handler = MockedHandler(msocket, ("127.0.0.1", 8888), BaseServer)
+        with patch.object(handler, "handle_request") as mock_handler:
+            getattr(handler, method)()
+            mock_handler.assert_called_once()
+
+
+def test_handle_favicon() -> None:
+    with patch("socket.socket") as msocket:
+        msocket.makefile = lambda a, b: io.BytesIO(b"GET /favicon.ico HTTP/1.1\r\n")
+        msocket.rfile.close = lambda: 0
+        handler = MyLambdaDevHandlerHelloWorld(msocket, ("127.0.0.1", 8888), BaseServer)
+    assert handler.handle_request() is None
+
+
+def test_handle_txt() -> None:
+    with patch.object(MyLambdaDevHandler, "_send_json") as mocked_send:
+        with patch("socket.socket") as msocket:
+            msocket.makefile = lambda a, b: io.BytesIO(b"GET /txt HTTP/1.1\r\n")
+            msocket.rfile.close = lambda: 0
+            MyLambdaDevHandlerHelloWorld(msocket, ("127.0.0.1", 8888), BaseServer)
+
+        mocked_send.assert_has_calls(
+            [
+                call(200, "HelloWorld", {"Content-Type": "text/plain"}),
+                call(
+                    500,
+                    {"error": "Server error"},
+                    headers={"Content-Type": "application/json;charset=UTF-8"},
+                ),
+            ]
+        )
+
 
-def test_my_dev_server():
-    dev_serv = MyDevServer(HelloWorld)
+def test_handle_json() -> None:
+    byte_string = (
+        b'POST /p HTTP/1.1\r\nContent-Length: 8\r\nContent-Type: application/json\r\n\r\n{"a": 1}'
+    )
+    with patch.object(MyLambdaDevHandler, "_send_json") as mocked_send:
+        with patch("socket.socket") as msocket:
+            msocket.makefile = lambda a, b: io.BytesIO(byte_string)
+            msocket.rfile.close = lambda: 0
+            MyLambdaDevHandlerHelloWorld(msocket, ("127.0.0.1", 8888), BaseServer)
+
+        mocked_send.assert_has_calls(
+            [
+                call(200, {"message": "uploaded"}, {"Content-Type": "application/json"}),
+                call(
+                    500,
+                    {"error": "Server error"},
+                    headers={"Content-Type": "application/json;charset=UTF-8"},
+                ),
+            ]
+        )
+
+
+def test_my_dev_server() -> None:
+    dev_serv = MyDevServer(MyLambdaDevHandlerHelloWorld)
     assert dev_serv.server_address == ("localhost", 8000)
     assert dev_serv.port == 8000
     assert dev_serv.address == "localhost"
     assert issubclass(dev_serv.my_handler, MyLambdaDevHandler)
+
+
+def test_my_dev_server_run() -> None:
+    dev_serv = MyDevServer(MyLambdaDevHandlerHelloWorld)
+    dev_serv.start()
+    assert dev_serv.is_alive()
```

Comment:
Don't you test that in a separate test module? `test_dev_server_running`? 🤔  I think I am getting a little lost. You are mixing the running server with the one which is partially mocked, and with that poor naming of the tests and duplicate part of the code, this won't be maintainable.

I wrote it before, but part of the tests for this library is more like integration tests which should be kept separately and better implemented/documented.